### DISCUSSION TITLE
Open screen in modal problem

### DISCUSSION THREAD

**Discussion by m7kael** [Budibase/budibase#10023] [Help] [answered]
Hi! 
I am having problems with opening a screen in modal but only in one of my apps. I have a couple of different apps where opening a screen in modal with the navigate to actions step works fine, but one of my apps it does not work. I have tried everything that i can think of but cannot figure it out. 

This is the action set for the button:
![Screenshot 2023-03-14 at 17 22 56](https://user-images.githubusercontent.com/112936118/225071521-dc26440a-5176-4380-8c37-b3a23e5dac78.png)

And i get this error in the javascript console:
![Screenshot 2023-03-14 at 17 22 34](https://user-images.githubusercontent.com/112936118/225071729-4785e343-ead3-425d-9789-179860b68226.png)

I have tried adding new screens, renaming the screen, defining the url in javascript, buttons and icons, exporting and importing the app again, you name it. But, if i put a "#" before the url like below, it navigates to the correct screen but does not open it in modal.

![Screenshot 2023-03-14 at 17 27 49](https://user-images.githubusercontent.com/112936118/225072601-f0555139-c7fa-4a79-a045-30b536cd29e2.png)

As i mentioned before this is only on one of my apps, all the other works as intended and the same if i create a new one. Anyone suffering from the same issues or any advice would be appriciated?

**Comment by ConorWebb96** [comment]
Hey @m7kael, 

This is very strange behaviour. The error suggests that the issue is to do with it escaping due to a special character being used somewhere. From the screenshot you provided it doesn't seem to be the case though? Could we get an app export so we can take a closer look?

Thanks!

**Reply by m7kael** [reply]
Hey @ConorWebb96,

I sent the app export to "support@budibase.com".

**Comment by melohagan** [comment]
Hey @m7kael I had a look at your app export. There's a lot of screens in there: which one has the button Navigate To action?

I did see it in `/home` but it doesn't really make sense there because you were trying to open the home screen in a modal within the home screen. 

The screens dropdown doesn't include a `/home` as an option on the Home screen.

**Reply by m7kael** [reply]
Yes so on the `/home` screen the button you are reffering to was the one i used for testing and problem solving, it does not make sense to open the same screen, but it was just an example

Another one is on the screen `/sales` and Create new

**Comment by melohagan** [comment, marked as answer]
Hey @m7kael 

I believe I've located the problem. 

I added a console log before the *Show Modal* function is called.  
![Screenshot 2023-03-15 at 14 03 44](https://user-images.githubusercontent.com/101575380/225332855-4c7124b0-64ea-43b2-8291-27e539bd1b7b.png)

It seems to take issue with the Euro!
![Screenshot 2023-03-15 at 14 08 40](https://user-images.githubusercontent.com/101575380/225334185-cd8b8b76-2cfa-48d6-87f7-f9bbfa679912.png)

As a test I replace the Euro with the USD '$' and it then worked:
![Screenshot 2023-03-15 at 14 09 32](https://user-images.githubusercontent.com/101575380/225334451-3b745889-f2b4-4cd5-a51f-a253065c8902.png)

It also worked when I encoded the euro symbol like so: `{{ encodeURI € }}`

There was a related issue opened a while ago here: https://github.com/Budibase/budibase/issues/7848

Unfortunately I believe we closed it because there wasn't much we could do about it at the time. I can see another user has recently commented on that issue so I'll re-open it to see if there's any quick win, or workaround.

**Reply by m7kael** [reply]
Hey @melohagan,
Many thanks!! I can confirm by deleteting the states the original app it now works as intended. I will re-think my solution and find a workaround, so no need to re-open it for now.

**Reply by benaunt** [reply]
We are also encountering this issue.

App was fine, set one state to persist, but did not save the change.

Now modals will not open, despite the state not being set to persist.

Thanks

**Comment by mincheol-jeong** [comment]
I'm having the same problem as above and I'm not able to solve it. It does not open as a modal window even if there is no emoticon used. Any solution?

**Comment by offbase-solutions** [comment]
Still awaiting a fix!!
